src/lib/storage/pos_list.hpp

```diff
@@ -53,6 +53,14 @@ struct PosList final : private pmr_vector<RowID> {
 
   PosList& operator=(PosList&& other) = default;
 
+  PosList copy() const {
+    PosList copy(this->size());
+    for (uint32_t idx = 0; idx < this->size(); ++idx) {
```

Comment:
Hm, could it make sense to use something like `std::copy` hear?

Reply:
I just executed a quick [benchmark code](https://gist.github.com/mweisgut/108ac25a594eaeb9fdf77955c2939e9e) and got [these results](https://gist.github.com/mweisgut/53bf004c7130d68f368745e0eee9d8c5) (executed on pella, release build, using clang-7).
Therefore I would say that the access [] variant is slightly faster than the std::copy variant. However, the std::copy variant would be more compact.

Reply:
Well, since the mentioned results show marginal differences between the access [] and std::copy variants and there are also iterations in which the std::copy variant is faster, we should probably use the standard lib.

Reply:
Thank you looking into it that deeply